### ISSUE TITLE
Make diffusion_policy sampler tensors respect configured device

### ISSUE THREAD

**Issue by ll7** [ll7/robot_sf_ll7#4181]
## Problem

Post-merge spot-check of #4156 found a device-consistency risk in the `diffusion_policy` runtime path.

`DiffusionPolicyAdapter.__init__()` moves the encoder and sampler to `self._device`, and `plan()` moves `node_features` / `mask` to that device before building the condition tensor. However, `DiffusionActionSampler.sample(...)` creates action and timestep tensors without a device argument, and the adapter always constructs `torch.Generator(device="cpu")`:

```python
actions = torch.zeros((num_samples, self.action_dim), dtype=torch.float32)
actions = torch.randn((num_samples, self.action_dim), generator=generator)
t = torch.full((num_samples, 1), float(step + 1) / float(denoising_steps))
```

For `DiffusionPolicyConfig.device` values such as `cuda`, `condition_batch` is on CUDA while `actions` and `t` remain on CPU. The subsequent `torch.cat([actions, t, condition_batch], dim=-1)` is therefore expected to fail with a device mismatch.

## Expected behavior

The diffusion-policy runtime should either:

- construct all sampler tensors on `condition.device` / sampler parameter device; and
- create or use a compatible generator for stochastic sampling on that device;

or fail closed early with a clear unsupported-device diagnostic.

## Acceptance criteria

- Keep CPU behavior unchanged.
- Ensure `actions`, `t`, and `condition_batch` are on the same device before concatenation.
- Add CPU regression coverage that asserts sampler tensor device follows the input condition device where possible.
- Add a CUDA smoke/regression test guarded by `pytest.mark.skipif(not torch.cuda.is_available(), ...)`, or an equivalent unit-level check that prevents reintroducing mixed-device sampling.
- Preserve the `allow_untrained_smoke` claim boundary; this issue is about runtime device correctness, not trained-policy evidence.

Refs #4156 and #4010.

**Comment by ll7** [comment]
## Implementation plan

### Boundary

This is a device-consistency fix for the diagnostic diffusion-policy runtime path from #4156 / #4010. Keep CPU behavior and the `allow_untrained_smoke` claim boundary unchanged.

### Phase 1 — Align sampler tensors to the condition device

Update `DiffusionActionSampler.sample(...)` in `robot_sf/planner/diffusion_policy.py` so every temporary tensor is created on `condition.device` and with a compatible dtype:

```python
device = condition.device
dtype = condition.dtype
actions = torch.zeros((num_samples, self.action_dim), dtype=dtype, device=device)
t = torch.full((num_samples, 1), value, dtype=dtype, device=device)
```

For random sampling, pass the same `device` to `torch.randn(...)`. Keep `condition_batch` on the same device before concatenation.

### Phase 2 — Fix generator construction

Replace the unconditional CPU generator in `DiffusionPolicyAdapter.__init__()` with a helper that creates a generator for `self._device` when supported. If the configured device is not supported by the installed PyTorch runtime, fail early with a clear message that names the configured device.

### Phase 3 — Preserve CPU determinism

Existing CPU behavior should stay deterministic for a fixed seed. Avoid moving tensors back to CPU until after sampling is complete and candidates are converted for guidance selection.

### Phase 4 — Tests

Add or extend diffusion-policy tests:

1. CPU deterministic sampling returns candidates on the same device as the input condition.
2. CPU random sampling works with the adapter-owned generator.
3. A guarded CUDA test runs only when `torch.cuda.is_available()` and verifies that `plan()` completes with `device="cuda"` and `allow_untrained_smoke=true`.
4. Diagnostics include the configured runtime device.

### Validation

```bash
uv run pytest tests -k "diffusion_policy" -q
uv run ruff check robot_sf/planner/diffusion_policy.py tests
BASE_REF=origin/main scripts/dev/pr_ready_check.sh
```

### Acceptance mapping

- CPU path remains unchanged.
- Sampler tensors and condition tensor share one device before concatenation.
- CUDA path is covered by a guarded test or fails closed with a clear device diagnostic.
- This remains runtime-correctness work, not trained-policy evidence.
